drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -246,8 +246,11 @@ impl Handler {
             }
             MsgType::UpdCurrentVersionReq => {
                 let version = self.update.current_version();
-                let rsp: Result<ImageVersion, u32> = Ok(version);
-                hubpack::serialize(tx_payload, &rsp)?
+                hubpack::serialize(tx_payload, &version)?
```

Comment:
The information we will need includes:
Current version in stage0,
Current version in A, // if this is known, B is unknown
Current version in B, // if this is known, A is unknown
Currently executing bank, // given by active_slot() which you have added.
Currently executing version, // given by current_version()
Next executing bank on reset. // unknown

current_version() tells you the version of the currently executing code but doesn't tell you which bank you are executing out of.

We should establish the APIs for these data early.

Reply:
Thank you for the details @lzrd. I will implement this in a follow up commit immediately.